spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -459,6 +459,7 @@ protected void actualStart() throws Exception {
 								}
 								this.logger.error("Error creating consumer; retrying in " + nextBackOff, e);
 								doShutdown();
+								this.consumersToRestart.clear();
```

Comment:
While I don't see any apparent issues with the solution I've implemented in this PR, it might be safer to use an `if(started)` switch in `doConsumeFromQueue`. 
Though, `org.springframework.amqp.rabbit.listener.DirectMessageListenerContainer.SimpleConsumer#cancelConsumer` does also add the current consumer to the list of consumers to restart. If this methods were invoked while the listener container is looping in its startup phase, there would still be a memory leak. So I suggest that we either wrap all `consumersToRestart.add()` calls in an `if(started)` or keep the `consumersToRestart.clear()` in the start loop.